src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Account/Pages/Manage/SetPassword.razor

```diff
@@ -60,6 +64,11 @@
 
     private async Task OnValidSubmitAsync()
     {
+        if (user is null)
+        {
+            throw new InvalidOperationException("User is not loaded; failed to set password.");
```

Comment:
We need to make this message more "human-friendly".

Alternatively, within `OnIntializedAsync` we can check RenderInfo.IsInteractive and potentially throw an exception there, or display an alternative UI with an error message (or redirect to the error page).

@MackinnonBuck since this affects the Identity UX it will be good if we can figure this out together.

Ultimately with the exception going away we need to provide an alternative, it might be throwing, redirecting to the error page or showing a message in this page.

Reply:
@MackinnonBuck just FYI, we might merge this to unblock things and do a follow up to cleanup things when we've had a chance to talk together through this.